### PR TITLE
Add the release-schedules endpoint

### PR DESCRIPTION
**Pull request by abompard**
Fixes: #294 #459

About issue #294: to know which releases are currently covered by a specific SLA, you can use the following query (using the SLA `bug_fixes` in this example):
`/rest_api/v1/release-schedules/?sla=bug_fixes&active=1&fields=release`

### CODE REVIEW

**Comment by abompard** [comment]
Thanks! I've rebased the commit and fixed the conflict.

**Comment by simozhan** [comment]
Thanks @abompard for pushing this new API. We can not provide the review comments currently, as we think it is not only for Dev's review, it also needs our QE's testing effort to merge this in internal instance. I am assuming this will not affect fedora's instance development. We will return back for merging these features in December 2017. If you want to prioritize the features from fedora, maybe you can discuss this in some meeting.

**Comment by pypingou** [comment]
Looks like there is a conflict on this now.

@simozhan do you know when this PR will be considered now?

**Comment by abompard** [comment]
I can resolve the conflict if necessary, it's probably very simple.

**Comment by pypingou** [comment]
@abompard if you have the time, let's do this so the PR is ready when @simozhan has some time to review it :)

File: pdc/apps/releaseschedule/serializers.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# Copyright (c) 2017 Red Hat
+# Licensed under The MIT License (MIT)
+# http://opensource.org/licenses/MIT
+#
+from rest_framework import serializers
+
+from pdc.apps.release.models import Release
+from pdc.apps.componentbranch.models import SLA
+from pdc.apps.common.fields import ChoiceSlugField
+from pdc.apps.common.serializers import DynamicFieldsSerializerMixin, StrictSerializerMixin
```
Comment (lubomir) [inline pdc/apps/releaseschedule/serializers.py]: These mixins should now be imported from `contrib.drf_introspection.serializers`.

File: pdc/apps/releaseschedule/serializers.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+from .models import ReleaseSchedule
+from . import signals
+
+
+class ReleaseScheduleSerializer(DynamicFieldsSerializerMixin,
+                                StrictSerializerMixin,
```
Comment (lubomir) [inline pdc/apps/releaseschedule/serializers.py]: Only `StrictSerializerMixin` should be used here, it inherits from `DynamicFieldsSerializerMixin` itself.

**Comment by abompard** [comment]
Thanks for the fixes @lubomir. About the signals, I've seen other components of PDC with signals in these methods, so I figured they are an internal convention and added them too. I can remove them if you prefer.

**Comment by lubomir** [comment]
The signals in `pdc.apps.release` are used by an internal extension. However I would say that turned out as not a very good design decision and I'd rather not repeat the same mistake. Please remove the signals.

**Comment by abompard** [comment]
Done! Thanks.

File: pdc/apps/releaseschedule/serializers.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    def update(self, instance, validated_data):
+        instance = super(ReleaseScheduleSerializer, self).update(instance, validated_data)
+        if hasattr(instance, 'pk'):
+            # reload to make sure changes in mapping are reflected
+            instance = ReleaseSchedule.objects.get(pk=instance.pk)
+        return instance
+
+    def create(self, validated_data):
+        instance = super(ReleaseScheduleSerializer, self).create(validated_data)
+        return instance
```
Comment (lubomir) [inline pdc/apps/releaseschedule/serializers.py]: The `create` and `update` methods can now be dropped.

File: pdc/apps/releaseschedule/signals.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#
```
Comment (lubomir) [inline pdc/apps/releaseschedule/signals.py]: This whole file can be removed as well.

**Pull request merged**